File: Elements/TextField.py
from Elements.BaseElement import BaseElements
from LogerUtils import Logger


class TextField(BaseElements):

    def send_keys(self, keys, clear=True):
        element = self.find_element_with_wait()
        if clear:
            element.clear()
            element.send_keys(keys)
        else:
            element.send_keys(keys)
        Logger.info('Input text to: ' + self.name)

File: elements/TextField.js
import BaseElement from "./BaseElement.js";
import Logger from '../utils/logUtils.js';

export default class TextField extends BaseElement{
    async sendKeys (value) {
        Logger.infoLog('Send value to element ' + this.name);
        $(this.locator).clearValue();
        return (await $(this.locator).setValue(value));
    };
};